eSPI: NPCX/ITE: Enable conditional virtual wire valid bit check

vphirema opened the pull request:
On the new Intel SoC, the "Valid" bit of the Virtual Wire is set only for Virtual Wires that undergo changes. This behavior differs from previous generations. Therefore, to maintain backward compatibility, a conditional check for the virtual wire valid bit is added for processing the virtual wire level.

vphirema commented:
cc: @MulinChao

andyross commented on drivers/espi/espi_npcx.c:
```diff
@@ -872,8 +872,11 @@ static int espi_npcx_receive_vwire(const struct device *dev,
 			val = GET_FIELD(inst->VWEVMS[reg_idx],
 							NPCX_VWEVMS_WIRE);
+
+#if defined(CONFIG_ESPI_VWIRE_VALID_BIT_CHECK)
```
Likewise IS_ENABLED() is preferred for simple tests that don't involve e.g. symbols that don't exist in certain conditions.

vphirema replied:
done

andyross commented on drivers/espi/espi_it8xxx2.c:
```diff
@@ -724,18 +724,23 @@ static int espi_it8xxx2_receive_vwire(const struct device *dev,
 	}
+#else
+	*level = !!(vw_reg->VW_INDEX[vw_index] & level_mask);
+#endif
```
Style nit, something like this is cleaner than #ifery for simple changes: 

```c
*level = 0;
if (IS_ENABLED(CONFIG_ESPI_VWIRE_VALID_BIT_CHECK) &&
    && vw_reg->VW_INDEX[vw_index] & valid_mask) {
	*level = !!(vw_reg->VW_INDEX[vw_index] & level_mask);
}
```

vphirema replied:
looks like i need to use this way
        if (IS_ENABLED(CONFIG_ESPI_VWIRE_VALID_BIT_CHECK)) {
                if (vw_reg->VW_INDEX[vw_index] & valid_mask) {
                        *level = !!(vw_reg->VW_INDEX[vw_index] & level_mask);
                } else {
                        *level = 0;
                }
        } else {
                *level = !!(vw_reg->VW_INDEX[vw_index] & level_mask);
        }

andyross reviewed (dismissed):
Nits but no complaints

MulinChao reviewed (approved):
LGTM for npcx part.

vphirema committed:
eSPI: NPCX/ITE: Enable conditional virtual wire valid bit check

On the new Intel SoC, the "Valid" bit of the Virtual Wire is set only for
Virtual Wires that undergo changes. This behavior differs from previous
generations. Therefore, to maintain backward compatibility, a conditional
check for the virtual wire valid bit is added for processing the virtual
wire level.

Signed-off-by: Vijay Hiremath <vijay.p.hiremath@intel.com>

vphirema commented:
Hi @albertofloyd , It's a power-on blocker. Could you please provide your feedback?

vphirema commented:
> > Hi @albertofloyd , It's a power-on blocker. Could you please provide your feedback?
> 
> @vphirema KConfig approach LGTM. Could you file an issue if not already done so this fix is eventually backported to 3.7?

Hi @albertofloyd  , here is the ticket I have filed. Do we need a separate one for backporting to 3.7
https://github.com/zephyrproject-rtos/zephyr/issues/81036
